presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -212,6 +213,23 @@ private Estimate calculateDistinctValuesCount(Map<String, PartitionStatistics> s
                 DoubleStream::max);
     }
 
+    private Estimate calculateDataSize(Map<String, PartitionStatistics> statisticsByPartitionName, String column, Estimate rowCount)
```

Comment:
I don't think we need `rowCount` (what is total row count) here. We need it for the number of nulls, because we are computing fraction (`totalNullsCount / totalRowsCount`). Total data size is a sum. The formula should be `sum(row_count_partition_N * avg_column_length_partition_N) over selected partitions`